### PR TITLE
Update contributing guide

### PR DESCRIPTION
**Pull request by mayank99**
Added info in CONTRIBUTING.md for:
- forking and cloning
- previewing demo html
- creating PR from fork
- `yarn createComponent`
- directory structure

Also fixed createComponent.js templates.

---

Markdown preview will be better at https://github.com/iTwin/iTwinUI/blob/mayank/update-contributing/CONTRIBUTING.md

### CODE REVIEW

**Review by FlyersPh9 — DISMISSED** [review]

**Review by bentleyvk — DISMISSED** [review]

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -2,17 +2,21 @@
 
 We welcome all types of contribution.
 
-Need a feature or found a bug? Please create an [issue.](https://github.com/iTwin/iTwinUI/issues)
+Need a feature or found a bug? Please create an [issue](https://github.com/iTwin/iTwinUI/issues).
 
-Want to contribute by creating a PR? Great! Then read further.
+Have a question or suggestion? Please create a [discussion](https://github.com/iTwin/iTwinUI/discussions).
+
+Want to contribute by creating a pull request? Great! [Fork the repo](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/working-with-forks) and read further.
```
Comment (veekeys) [inline CONTRIBUTING.md:9]: If you changed PR, then I guess go repository too.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -2,17 +2,21 @@
...
+
+---
 
 ## How to setup
 
-To clone and build iTwinUI you'll need [Git](https://git-scm.com) and [Yarn](https://yarnpkg.com) installed on your computer. From your command line:
+To clone and build iTwinUI, you'll need [Git](https://git-scm.com) and [Yarn 1](https://yarnpkg.com/getting-started/install) installed on your computer.
```
Comment (veekeys) [inline CONTRIBUTING.md:15]: Why Yarn 1?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -44,52 +50,45 @@ Before running this command make sure Docker is running. Read more about [tests]
...
-  - Example from `src/toggle-switch/classes.scss`, where index imports all mixins and relevant scss.
-
-    ```scss
-    @import './index';
-
-    .iui-toggle-switch {
-      @include iui-toggle-switch;
-    }
-    ```
+If you'd like to get your component added to iTwinUI, follow these guidelines.
```
Comment (veekeys) [inline CONTRIBUTING.md:63]: I feel here should not be used 'would'... 
image

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -44,52 +50,45 @@ Before running this command make sure Docker is running. Read more about [tests]
...
+- To approve test images, run `yarn approve`.
 
- `yarn test --filter="test_name"`
+- To delete old/unused tests images, run `yarn clean:images`.
 
-To approve tests use command:
+#### How to write tests:
 
-`yarn approve`
+- Create `.html` file in `backstop/tests` displaying the elements you wish to test and their all possible states, using the built CSS in `lib/css/`.
```
Comment (veekeys) [inline CONTRIBUTING.md:90]: I think file will be created already with script. So just need to write the html in there (start with display the element, i.e.)

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -117,32 +116,41 @@ How to write tests:
...
 
-### Committing your work
+---
 
-Before creating a pull request, make sure your changes address a specific issue. Do a search to see if there are any existing items that are still open. If you don't find one, you can create one.
+## Committing your work
+
+Before creating a pull request, make sure your changes address a specific issue. Do a search to see if there are any existing issues that are still open. If you don't find one, you can create one.
 
 To enable us to quickly review and accept your pull requests, always create one pull request per issue. Never merge multiple requests in one unless they have the same root cause. Be sure to follow best practices and keep code changes as small as possible. Avoid pure formatting changes or random "fixes" that are unrelated to the linked issue.
```
Comment (veekeys) [inline CONTRIBUTING.md:143]: Kinda repeating with whats written above

File: CONTRIBUTING.md (hunk 3 above)
Comment (mayank99) [reply CONTRIBUTING.md:63]: what 😄

File: CONTRIBUTING.md (hunk 2 above)
Comment (mayank99) [reply CONTRIBUTING.md:15]: because we use yarn 1, not yarn 2

**Review by FlyersPh9 — APPROVED** [review]

**Pull request merged**